server/lib/budget.js

```diff
@@ -367,33 +670,50 @@ export async function sumCollectivesTransactions(
   // Initialize totals
   if (ids) {
     for (const CollectiveId of ids) {
-      totals[CollectiveId] = { CollectiveId, currency: 'USD', value: 0 };
+      totals[CollectiveId] = { CollectiveId, currency: 'USD', value: 0, timeSeries: {} };
     }
   }
 
-  const results = await models.Transaction.findAll({
-    attributes: [
-      'CollectiveId',
-      groupBy,
-      [sequelize.fn('COALESCE', sequelize.fn('SUM', sequelize.col('amount')), 0), 'amountInCollectiveCurrency'],
-      [
-        sequelize.fn('COALESCE', sequelize.fn('SUM', sequelize.col('netAmountInCollectiveCurrency')), 0),
-        'netAmountInCollectiveCurrency',
-      ],
-      [sequelize.fn('COALESCE', sequelize.fn('SUM', sequelize.col('amountInHostCurrency')), 0), 'amountInHostCurrency'],
-      [
-        sequelize.fn(
-          'COALESCE',
-          sequelize.literal(
-            'SUM(COALESCE("amountInHostCurrency", 0)) + SUM(COALESCE("platformFeeInHostCurrency", 0)) + SUM(COALESCE("hostFeeInHostCurrency", 0)) + SUM(COALESCE("paymentProcessorFeeInHostCurrency", 0)) + SUM(COALESCE("taxAmount" * "hostCurrencyFxRate", 0))',
-          ),
-          0,
+  const CollectiveId = includeChildren
+    ? [
+        sequelize.fn('COALESCE', sequelize.col('collective.ParentCollectiveId'), sequelize.col('collective.id')),
+        'CollectiveId',
+      ]
+    : sequelize.col('CollectiveId');
+
+  const attributes = [
+    CollectiveId,
+    currencyField,
+    [sequelize.fn('COALESCE', sequelize.fn('SUM', sequelize.col('amount')), 0), 'amountInCollectiveCurrency'],
+    [
+      sequelize.fn('COALESCE', sequelize.fn('SUM', sequelize.col('netAmountInCollectiveCurrency')), 0),
+      'netAmountInCollectiveCurrency',
+    ],
+    [sequelize.fn('COALESCE', sequelize.fn('SUM', sequelize.col('amountInHostCurrency')), 0), 'amountInHostCurrency'],
+    [
+      sequelize.fn(
+        'COALESCE',
+        sequelize.literal(
+          'SUM(COALESCE("amountInHostCurrency", 0)) + SUM(COALESCE("platformFeeInHostCurrency", 0)) + SUM(COALESCE("hostFeeInHostCurrency", 0)) + SUM(COALESCE("paymentProcessorFeeInHostCurrency", 0)) + SUM(COALESCE("taxAmount" * "hostCurrencyFxRate", 0))',
         ),
-        'netAmountInHostCurrency',
-      ],
+        0,
+      ),
+      'netAmountInHostCurrency',
     ],
+  ];
+
+  const group = [CollectiveId, currencyField];
+
+  if (withTimeSeries) {
+    attributes.push([sequelize.fn('DATE_TRUNC', timeUnit, sequelize.col('Transaction.createdAt')), 'date']);
+    group.push('date');
```

Comment:
Make it generic so we can be able to group by something else like `FromCollectiveId`.

Reply:
Fixed!